docs/pct/README.md

```diff
@@ -0,0 +1,291 @@
+
+### Structure
+
+A PCT must contain a `pct-config.yml` in the root directory, alongside a `content` directory. The root directory must be named the same as the `id` value defined in  your `pct-config.yml`
+
+The `content` directory contains the files and folders required to produce the `project` or `item`.
+
+To mark a file as a template, use the `.tmpl` extension. Templated files can also use the global variable of `{{pct_name}}` to access the input from the `--name` cli argument.
+
+> :memo: Folders within the `content` directory can also use the `{{pct_name}}` variable
+
+Example template file names:
+
+``` bash
+myConfig.json.tmpl
+{{pct_name}}_spec.rb
+```
+
+> :memo: One, all or none of the files can be templated.
+
+#### pct-config.yml
+
+Format of pct-config.yml
+
+``` yaml
+---
+template:
+  id: <a unique name>
+  type: <'item' or 'project'>
+  display: <a human readable name>
+  version: <semver>
+  url: <url to project repo>
+
+<template parameters>
+```
+
+> :memo: Template `id` must not contain spaces or special characters. We recommend using a hyphen to break up the identifier.
+
+Example pct-config.yml:
+
+``` yaml
+---
+template:
+  id: example-template
+  type: project
+  display: Example
+  version: 0.1.0
+  url: https://github.com/puppetlabs/pct-example
+```
+
+Example structure for `example-template`:
+
+``` bash
+> tree ~/.pdk/pct/example-template-params
+/Users/me/.pdk/pct/example-template-params
+├── content
+│   └── example.txt.tmpl
+└── pct-config.yml
+```
+
+Example pct-config.yml with parameters:
+
+``` yaml
+---
+template:
+  id: example-template-params
+  type: project
+  display: Example with Parameters
+  version: 0.1.0
+  url: https://github.com/puppetlabs/pct-example-params
+
+
+example_template:
+  foo: "bar"
+  isPuppet: true
+  colours:
+  - "Red"
+  - "Blue"
+  - "Green"
+
+```
+
+### Templating Language
+
+PCT uses [Go's templating language](https://golang.org/pkg/text/template/#hdr-Actions).
+
+In the above template `example-template-params` the parameters can be accessed in a `.tmpl` file like so:
+
+``` go
+{{.example_template.foo}}
+{{.example_template.isPuppet}}
+{{.example_template.colours}}
+```
+
+outputs:
+
+``` text
+bar
+<no value>
+[Red Blue Green]
+```
+
+As a template author you can chose your own parameters and parameter structure so long as it is [valid YAML](https://yaml.org/spec/1.2/spec.html). Then utilise the GO templating language to display or iterate over these.
+
+For most templates, we believe that you can do most of the things you would want with these common template controls:
+
+``` go
+// Outputs the value of `foo` defined within pct.yml
+{{.example_template.foo}}
+
+// A conditional
+{{if .example_template.isPuppet}}
+ "boo :("
+{{else}}
+ "yay!"
+{{end}}
+
+// Loops over all "colours" and renders each using {{.}}
+{{range .example_template.colours}} {{.}} {{end}}
+```
+
+For more examples look at the existing templates provided at `$HOME/.pdk/pct/`
+
+### Testing
```

Comment:
I'd rather not include the section than leave it empty. I thought the markdown rendered incorrectly instead of it being empty 😂

Reply:
This was more of a holder to ask the question of how to best test a template.